upstairs/src/dummy_downstairs_tests.rs

```diff
@@ -711,4 +722,999 @@ pub(crate) mod protocol_test {
+                .await
+                .send(Message::FlushAck {
+                    upstairs_id: harness.guest.get_uuid().await.unwrap(),
+                    session_id: harness
+                        .ds3
+                        .upstairs_session_id
+                        .lock()
+                        .await
+                        .unwrap(),
+                    job_id: flush_job_id,
+                    result: Ok(()),
+                })
+                .await
+                .unwrap();
+
+            // Wait for the flush to come back
+            jh.await.unwrap();
+        }
+
+        // Send ds1 responses for the jobs it saw
+        for (i, job_id) in job_ids.iter().enumerate().take(100) {
+            match harness
+                .ds1()
+                .await
+                .fw
+                .lock()
+                .await
+                .send(Message::ReadResponse {
+                    upstairs_id: harness.guest.get_uuid().await.unwrap(),
+                    session_id: harness
+                        .ds1()
+                        .await
+                        .upstairs_session_id
+                        .lock()
+                        .await
+                        .unwrap(),
+                    job_id: *job_id,
+                    responses: Ok(vec![make_blank_read_response()]),
+                })
+                .await
+            {
+                Ok(()) => {
+                    info!(
+                        harness.log,
+                        "sent read response for job {} = {}", i, job_id,
+                    );
+                }
+
+                Err(e) => {
+                    // We should be able to send a few, but at some point
+                    // the Upstairs will disconnect us.
+                    error!(
+                        harness.log,
+                        "could not send read response for job {} = {}: {}",
+                        i,
+                        job_id,
+                        e
+                    );
+                    break;
+                }
+            }
+        }
+
+        // Assert the Upstairs isn't sending ds1 more work, because it is
+        // Faulted
+        assert!(matches!(ds1_messages.try_recv(), Err(TryRecvError::Empty)));
+
+        // Reconnect ds1
+        drop(ds1_messages);
+        jh1.abort();
+
+        let ds1 = harness.take_ds1().await;
+        let ds1 = ds1.close();
+        let ds1 = ds1.into_connected_downstairs().await;
+
+        ds1.negotiate_start().await;
+        ds1.negotiate_step_extent_versions_please().await;
+
+        let (_jh1, mut ds1_messages) = ds1.spawn_message_receiver().await;
+
+        // The Upstairs will start sending LiveRepair related work, which may be
+        // out of order. Buffer some here.
+
+        let mut ds1_buffered_messages = vec![];
+        let mut ds2_buffered_messages = vec![];
+        let mut ds3_buffered_messages = vec![];
+
+        for eid in 0..10 {
```

Comment:
Is the 10 because we have 10 extents in our test downstairs?

Reply:
Yep